x-pack/plugins/ml/public/file_datavisualizer/components/edit_flyout/overrides.js

```diff
@@ -0,0 +1,337 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+
+import React, {
+  Component,
+} from 'react';
+
+import {
+  EuiForm,
+  EuiFormRow,
+  EuiFieldText,
+  EuiSuperSelect,
+  EuiCheckbox,
+  EuiSpacer,
+  EuiTitle,
+} from '@elastic/eui';
+
+import {
+  formatOptions,
+  timestampFormatOptions,
+  delimiterOptions,
+  quoteOptions,
+  charsetOptions,
+} from './options';
+
+export class Overrides extends Component {
+  constructor(props) {
+    super(props);
+
+    this.originalSettings = this.props.originalSettings;
+    this.overrides = this.props.overrides;
+
+    const {
+      charset,
+      format,
+      hasHeaderRow,
+      columnNames,
+      delimiter,
+      quote,
+      shouldTrimFields,
+      grokPattern,
+      timestampField,
+      timestampFormat,
+    } = this.overrides;
```

Comment:
Similar to the above, not sure `this.overrides = ...` in necessary, could be just `} = this.props.overrides;` here.